### PR TITLE
[PS-1906] Extract uploadEvents from EventService into new service

### PR DESCRIPTION
**Pull request by djsmith85**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
This PR is prep work for an easier integration of `event.service` for MV3 which will use the AlarmsApi to uploadEvents on an interval. Which will get implemented with https://github.com/bitwarden/clients/pull/4103. The idea here was to split up the responsibility of collection and upload of events, which make service orchestration/instantiation easier. 

## Code changes
1st commit
Move abstractions/event.service to abstractions/**event**/event.service
Move services/event.service to services/**event**/event.service
Fix all the imports

2nd commit
Move `uploadEvents` from `EventService` to `EventUploadService`
Create event-upload-service-factory
Fix wiring up all the dependencies

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: libs/common/src/abstractions/event/event.service.ts [hunk 1]
```diff
@@ -7,6 +7,5 @@ export abstract class EventService {
     uploadImmediately?: boolean,
     organizationId?: string
   ) => Promise<any>;
-  uploadEvents: (userId?: string) => Promise<any>;
   clearEvents: (userId?: string) => Promise<any>;
```
Comment (MGibson1) [inline libs/common/src/abstractions/event/event.service.ts:10]: `clearEvents` is only called in the context of uploading them, this should be part of the upload service, and as you have done, shouldn't need to be public, just part of clear events.

File: libs/common/src/abstractions/event/event-upload.service.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export abstract class EventUploadService {
```
Comment (MGibson1) [inline libs/common/src/abstractions/event/event-upload.service.ts:1]: IMO `EventService` should also be renamed to `EventCollectionService`

**Review by MGibson1 — CHANGES_REQUESTED** [review]

File: libs/common/src/abstractions/event/event.service.ts (hunk 1 above)
Comment (djsmith85) [reply libs/common/src/abstractions/event/event.service.ts]: I have removed clearEvents-method from the abstract as it was only needed for logout-procedures of the individual apps. The uploadEvents in those take care of clearing the state anyway. Changes in e5a587e

File: libs/common/src/abstractions/event/event-upload.service.ts (hunk 2 above)
Comment (djsmith85) [reply libs/common/src/abstractions/event/event-upload.service.ts:1]: A small comment, with a potentially high impact on changed files, which most reviewers regret 😁

- `EventService` has been renamed to `EventCollectionService` (9b33677).
- The init method from `EventCollectionService` got moved to `EventUploadService` (a5b8fc2)

**Review by MGibson1 — APPROVED** [review]
Nice

**Pull request merged**